blsSignatures/blsSignatures.go

```diff
@@ -43,9 +45,13 @@ func GenerateKeys() (PublicKey, PrivateKey, error) {
 }
 
 func PublicKeyFromPrivateKey(privateKey PrivateKey) (PublicKey, error) {
-	pubKey := &bls12381.PointG2{}
-	g2 := bls12381.NewG2()
-	g2.MulScalar(pubKey, g2.One(), privateKey)
+	pubKey := new(bls12381.G2Affine)
+	g2 := new(bls12381.G2Affine)
+	g2.X.SetOne()
+	g2.Y.SetOne()
```

Comment:
Here, and in a few other places, the G2 point with coordinates (1, 1) is used to replace `g2.One()` from the old library. However, that's not the same thing. `g2.One()` returned the basepoint (aka generator). To get it in the new library, you can call `bls12381.Generators()` and use the last return value (it's basically free to call as it just returns constants).

Reply:
Done.